### PR TITLE
DS Picker: first item is not active when filtering

### PR DESCRIPTION
**Pull request by ivanortegaalba**
**Problem**
When opening the dropdown, the first item should be always active to use enter. When filtering, since the list of elements changes, it needs to recompute the styles.

**Solution**
Monitor the number of items in the list, and update the HTML attributes for the list items.

|Before|After|
|-|-|
|

Related to https://github.com/grafana/grafana/issues/66916

### CODE REVIEW

File: public/app/features/datasources/components/picker/DataSourceList.tsx [hunk 1]
```diff
@@ -54,7 +54,6 @@ export function DataSourceList(props: DataSourceListProps) {
   const styles = getStyles(theme, selectedItemCssSelector);
 
   const { className, current, onChange, enableKeyboardNavigation, onClickEmptyStateCTA } = props;
-  // QUESTION: Should we use data from the Redux store as admin DS view does?
```
Comment (ivanortegaalba) [inline public/app/features/datasources/components/picker/DataSourceList.tsx:57]: No, we should not. That endpoint returns the details of the Ds, not the DS settings.

**Comment by ivanortegaalba** [comment]
I'm applying the proposed changes

**Comment by ivanortegaalba** [comment]
@Elfo404 I applied the changes to the hook to avoid re-rendering when using arrows. The re-renders when opening still happening

File: public/app/features/datasources/hooks.ts [hunk 2]
```diff
@@ -60,6 +60,7 @@ export function useDatasource(dataSource: string | DataSourceRef | DataSourceIns
 export interface KeybaordNavigatableListProps {
   keyboardEvents?: Observable<React.KeyboardEvent>;
   containerRef: React.RefObject<HTMLElement>;
+  numberOfItems?: number;
```
[the remark this answers is not in the document]
Comment (ivanortegaalba) [reply public/app/features/datasources/hooks.ts:63]: 😢

[the remark this answers is not in the document]
Comment (Elfo404) [reply public/app/features/datasources/hooks.ts:63]: lol

Comment (ivanortegaalba) [inline public/app/features/datasources/hooks.ts:63]:
```suggestion
```

**Review by Elfo404 — APPROVED** [review]

**Comment by Elfo404** [comment]
on thing: if this was a bug in 10.0.1 hen it should probably be mentioned in the changelog

**Pull request merged**